apps/mobile/src/components/Camera/QrCamera.tsx

```diff
@@ -0,0 +1,202 @@
+import { Camera, useCodeScanner, useCameraDevice, Code, CameraPermissionStatus } from 'react-native-vision-camera'
+import { View, Theme, H3 } from 'tamagui'
+import { Dimensions, Linking, Pressable, StyleSheet } from 'react-native'
+import React, { useCallback } from 'react'
+import { useRouter } from 'expo-router'
+
+const { width } = Dimensions.get('window')
+import { BlurView } from 'expo-blur'
+import { SafeFontIcon } from '@/src/components/SafeFontIcon'
+import { Badge } from '@/src/components/Badge'
+import { SafeButton } from '@/src/components/SafeButton'
+
+type QrCameraProps = {
+  heading?: React.ReactNode
+  footer: React.ReactNode
+  onScan: (code: Code[]) => void
+  isCameraActive: boolean
+  permission: CameraPermissionStatus
+}
+export const QrCamera = ({ heading = 'Scan a QR Code', footer, onScan, isCameraActive, permission }: QrCameraProps) => {
+  const router = useRouter()
+  const device = useCameraDevice('back')
+
+  const codeScanner = useCodeScanner({
+    codeTypes: ['qr'],
+    onCodeScanned: (codes) => {
+      onScan(codes)
+    },
+  })
+
+  const openSettings = useCallback(async () => {
+    await Linking.openSettings()
+  }, [])
+
+  const denied = permission === 'denied'
+
+  return (
+    <Theme name={'dark'}>
+      <View style={styles.container}>
+        {/* Camera View */}
+        {device && (
+          <Camera style={StyleSheet.absoluteFill} device={device} isActive={isCameraActive} codeScanner={codeScanner} />
+        )}
+
+        {/* Overlay with blurred edges */}
+        <View style={styles.overlay}>
+          <View flex={1}>
+            <BlurView
+              style={[styles.blurTop, denied && { backgroundColor: 'rgba(0, 0, 0, 1)' }]}
+              intensity={30}
+              // experimentalBlurMethod={'dimezisBlurView'}
+              tint={'systemUltraThinMaterialDark'}
+            >
+              <View style={styles.topContainer}>
+                <View style={{ flex: 1, marginTop: 30, marginLeft: 20, flexDirection: 'row' }}>
+                  <Pressable
+                    onPress={() => {
+                      router.back()
+                    }}
+                  >
+                    <Badge
+                      themeName="badge_background"
+                      circleSize="$9"
+                      content={<SafeFontIcon size={20} name="close" />}
+                    />
+                  </Pressable>
+                </View>
+
+                <View flex={1} justifyContent={'flex-end'} alignItems={'center'} marginBottom={'$8'}>
+                  {typeof heading === 'string' ? <H3>{heading}</H3> : heading}
+                </View>
+              </View>
```

Comment:
Can we move it to be inside a separated dumb component? Maybe CameraHeader or something like that